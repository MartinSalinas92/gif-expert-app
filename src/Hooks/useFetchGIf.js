import { useState, useEffect } from 'react';
import { getGif } from '../Helpers/getGif';


export const useFetchGifs = (category)=>{

    const [real, realCambio] = useState({

        data:[],
        loading:true
    });

      useEffect(() => {
        
        getGif(category)
            .then(img=>{
                console.log(img)

                setTimeout(() => {

                    realCambio({

                    
                        data:img,
                        loading:false
    
                    })
                    
                }, 3000);
               
            })
    }, [category])

    return real;
}