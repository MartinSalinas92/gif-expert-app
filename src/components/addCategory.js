
import React, {useState} from 'react'
import PropTypes from 'prop-types';

 const AddCategory = ({setCategories}) => {

    const [state, setstate] = useState('');

    const cambioTexto= (e)=>{

        //console.log(e.target.value);

        setstate(e.target.value)


    }

    const pushEnter= (e)=>{

        e.preventDefault();

        console.log(pushEnter);

        if(state.trim().length > 2){

            setCategories(cats => [state, ...cats])

        }

      

       
    }


    return (
        <form onSubmit={pushEnter}>
            <p>{state}</p>
                <input
                type="text" 
                value={state}
                onChange={cambioTexto}
                
                
                />
            
        </form>
    )
}

AddCategory.propTypes={

    setCategories: PropTypes.func.isRequired

}
export default AddCategory


