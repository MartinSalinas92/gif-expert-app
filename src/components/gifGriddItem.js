import React from 'react';
import PropTypes from 'prop-types';

export function GifGriddItem({img, title, images}) {

   
    return (
        <div className="card animate__animated animate__bounce">
             <img src={images} alt={title} />

             <p className="animate__animated animate__fadeInLeft">{title}</p>
            
            
        </div>
    )
}

GifGriddItem.propTypes={

    title:PropTypes.string.isRequired,
    images:PropTypes.string.isRequired
}





