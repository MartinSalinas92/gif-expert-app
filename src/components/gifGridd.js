import React from 'react';
import { useFetchGifs } from '../Hooks/useFetchGIf';
import { GifGriddItem } from './gifGriddItem';
import PropTypes from 'prop-types';

export const GifGridd = ({category}) => {

    

   const {data, loading}= useFetchGifs(category);

   
  

  

  

   

    return (
        <>
            <h3>{category}</h3>



           
                
            <div className="card-grid">

                {loading && <p>loading</p>}

            {
                data.map(img=>(

                    <GifGriddItem
                      key={img.id}  

                      {...img}

                    
                    />
                    
                ))
            }
                  
            
                
                    
            </div>
                    
                
           
           
        </>
    )
}

GifGridd.propTypes={

    category:PropTypes.string.isRequired

}
