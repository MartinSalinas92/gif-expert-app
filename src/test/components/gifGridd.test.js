import React from 'react';
import { shallow } from "enzyme";
import { GifGridd } from "../../components/gifGridd";
import { useFetchGifs } from '../../Hooks/useFetchGIf';
jest.mock('../../Hooks/useFetchGIf');

describe('Pruebas de componentes <GifGridd />' , ()=>{

    const category="Hulk";

    test('debe mostrarse correctamente', () => {

        useFetchGifs.mockReturnValue({
            data:[],
            loading:true


        });

        const wrapper= shallow(<GifGridd  category={category} />)

        expect(wrapper).toMatchSnapshot();
        
    });
    test('debe de mostrar items cuando se cargan imagenes useFetchGifs', () => {

        const gifs=[{
            id:'ABC',
            url:'http://localhost/cualquier/cosa.jpg',
            title:'cualquier cosa'
        }]


        useFetchGifs.mockReturnValue({
            data:gifs,
            loading:false


        });
        
        const wrapper= shallow(<GifGridd  category={category} />)

       // expect(wrapper).toMatchSnapshot();

        expect(wrapper.find('p').exists()).toBe(false);
        expect(wrapper.find('GifGriddItem').length).toBe(gifs.length)

        
        
    })
    
    
})