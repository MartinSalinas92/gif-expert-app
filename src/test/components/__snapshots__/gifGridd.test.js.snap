// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas de componentes <GifGridd /> debe mostrarse correctamente 1`] = `
<Fragment>
  <h3>
    Hulk
  </h3>
  <div
    className="card-grid"
  >
    <p>
      loading
    </p>
  </div>
</Fragment>
`;
